grpclb/src/main/java/io/grpc/grpclb/GrpclbLoadBalancer.java

```diff
@@ -114,11 +105,10 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
     newLbAddressGroups = Collections.unmodifiableList(newLbAddressGroups);
     List<EquivalentAddressGroup> newBackendServers =
         Collections.unmodifiableList(resolvedAddresses.getAddresses());
-    Map<String, ?> rawLbConfigValue = attributes.get(ATTR_LOAD_BALANCING_CONFIG);
-    Mode newMode = retrieveModeFromLbConfig(rawLbConfigValue, helper.getChannelLogger());
-    if (!mode.equals(newMode)) {
-      mode = newMode;
-      helper.getChannelLogger().log(ChannelLogLevel.INFO, "Mode: " + newMode);
+    GrpclbConfig newConfig = (GrpclbConfig) resolvedAddresses.getLoadBalancingPolicyConfig();
+    if (config == null || !config.equals(newConfig)) {
```

Comment:
If `config == null`, don't you want to use `GrpclbConfig.create(Mode.ROUND_ROBIN)`? This was previously handled in the retrieveModeFromLbConfig method. It doesn't look like `this.config` should ever be `null`, since `GrpclbState` can't handle a null config.

Reply:
thanks, done.